### ISSUE TITLE
Use puppeteer for @vx/text tests

### ISSUE THREAD

**Issue by techniq** [hshoff/vx#216]
Now that jest has support for [puppeteer](http://facebook.github.io/jest/docs/en/puppeteer.html#content), we need to use it for @vx/text [tests](https://github.com/hshoff/vx/blob/master/packages/vx-text/test/Text.test.js) as it requires a DOM to perform the text measurements (something that jsdom does not support which jest uses by default).

Here is a full working example: https://github.com/xfumihiro/jest-puppeteer-example

**Comment by montogeek** [comment]
I could contribute with this, You mean run all tests over Puppeteer instead of JSDom?

**Comment by techniq** [comment]
@montogeek that would be awesome!  We need to use puppeteer for @vx/text tests so it can returned the [measured width](https://github.com/hshoff/vx/blob/master/packages/vx-text/src/util/getStringWidth.js#L21) which is unavailable in JSDom (since there isn't a true DOM).

@hshoff I've been using [react-testing-library](https://github.com/kentcdodds/react-testing-library) over enzyme nowaways in my libraries ([example](https://github.com/techniq/react-fetch-component/blob/master/src/Fetch.test.js)), initially due to it not supporting React 16 APIs, but also has a simpler / easier to understand API (felt less magic-y to me).  Should we move to it as well?

**Comment by jackdbd** [comment]
I'm using puppeteer + jest + [jest-image-snapshot](https://github.com/americanexpress/jest-image-snapshot) for a project of mine. Here is what the code looks like:

```js
const puppeteer = require("puppeteer");
const HOST = "http://localhost:3000";
const EXPECTED_IMAGES = "src/__tests__/expected_images";

const directNavigationOptions = {
  waitUntil: "networkidle2",
};

describe("Visual regressions)", () => {
  // use the same browser instance for all visual regression tests to save time
  let browser = null;
  beforeAll(async () => {
    // WebGL context is available in headless mode too!
    browser = await puppeteer.launch({ headless: true });
  });
  afterAll(async () => {
    browser.close();
  });
  it("matches the expected snapshot of /", async () => {
    const page = await browser.newPage();
    const url = `${HOST}/`;
    await page.goto(url, directNavigationOptions);
    const options = {
      path: `${EXPECTED_IMAGES}/home.png`,
    };
    const image = await page.screenshot(options);
    expect(image).toMatchImageSnapshot();
  });
```

As you can see it's very similar to the one described in [this article](https://codeburst.io/automatic-visual-regression-testing-23cc06471dd) ([Navalia](https://github.com/joelgriffith/navalia) is no longer maintained).

**Comment by williaster** [comment]
closing this as unit tests were re-enabled / fixed in #790
